src/dao/api.dao.ts

```diff
@@ -247,70 +247,94 @@ export class NodeDao {
-      LEFT JOIN "Edge" as e2 ON n2.id = e2."startNodeId"
-      LEFT JOIN "Node" as n3 ON e2."endNodeId" = n3.id
-      LEFT JOIN "ClaimData" as cd ON c.id = cd."claimId"
-      LEFT JOIN "Image" as i ON c.id = i."claimId"
-      ${whereClause}
-      ORDER BY c."effectiveDate" DESC
-      LIMIT ${limit}
-      OFFSET ${offset}
-    `;
-
-    console.log("Feed entries raw query: " + rawQ.sql + "\n with values \n" + rawQ.values)
-    const feedEntries = await prisma.$queryRaw<FeedEntry[]>`${rawQ}`;
-    return feedEntries;
+    try {
+      if (typeof search === "string") {
+        search = decodeURIComponent(search);
+      }
+
+      const baseWhereClause = `
+        WHERE n1."entType" != 'CLAIM'
+          AND e.label != 'source'
+          AND c."effectiveDate" IS NOT NULL
+          AND n1.name IS NOT NULL
+          AND n1.name != ''
+      `;
+
+      const searchWhereClause = search
+        ? `
+            AND (
+              c.statement ILIKE '%${search}%' OR
+              c."sourceURI" ILIKE '%${search}%' OR
+              c."subject" ILIKE '%${search}%' OR
+              n1.name ILIKE '%${search}%' OR
+              n3.name ILIKE '%${search}%' OR
+              n3."descrip" ILIKE '%${search}%'
+            )
+          `
+        : "";
+
+      const whereClause = baseWhereClause + searchWhereClause;
+
+      const rawQ = Prisma.sql`
+        SELECT
+          n1.name as name,
+          n1.thumbnail as thumbnail,
+          n1."nodeUri" as link,
+          n1."descrip" as description,
+          c.id as claim_id,
+          c.statement as statement,
+          c.stars as stars,
+          c.score as score,
+          c.amt as amt,
+          c."effectiveDate" as effective_date,
+          c."howKnown" as how_known,
+          c.aspect as aspect,
+          c.confidence as confidence,
+          e.label as claim,
+          e2.label as basis,
+          n3.name as source_name,
+          n3.thumbnail as source_thumbnail,
+          n3."nodeUri" as source_link,
+          n3."descrip" as source_description
+        FROM "Claim" c
+        INNER JOIN "Edge" AS e ON c.id = e."claimId"
+        INNER JOIN "Node" AS n1 ON e."startNodeId" = n1.id
+        INNER JOIN "Node" AS n2 ON e."endNodeId" = n2.id
+        LEFT JOIN "Edge" as e2 ON n2.id = e2."startNodeId"
+        LEFT JOIN "Node" as n3 ON e2."endNodeId" = n3.id
+        LEFT JOIN "Image" as i ON c.id = i."claimId"
+        LEFT JOIN "ClaimData" as cd ON c.id = cd."claimId"
+        ${Prisma.raw(whereClause)}
+        ORDER BY c."effectiveDate" DESC
+        LIMIT ${limit}
+        OFFSET ${offset}
+      `;
+
+      const feedEntries = await prisma.$queryRaw<FeedEntry[]>(rawQ);
+
+      const claimIdMap = new Map<number, FeedEntry>();
+      feedEntries.forEach((entry) => {
+        if (!claimIdMap.has(entry.claim_id)) {
+          claimIdMap.set(entry.claim_id, entry);
+        }
+      });
+
+      const uniqueEntriesByClaimId = Array.from(claimIdMap.values());
```

Comment:
this 100% should not be necessary i don't understand why this is here, you can only have 1 claim id, something must be wrong with the query